pkg/flare/sysprobe/archive_linux.go

```diff
@@ -3,36 +3,94 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2024-present Datadog, Inc.
 
-//go:build process && linux
+//go:build linux
 
 package sysprobe
 
-import "github.com/DataDog/datadog-agent/pkg/process/net"
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	sysprobeclient "github.com/DataDog/datadog-agent/cmd/system-probe/api/client"
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+)
 
 // GetSystemProbeConntrackCached queries conntrack/cached, which uses our conntracker implementation (typically ebpf)
 // to return the list of NAT'd connections
-func GetSystemProbeConntrackCached(socketPath string) ([]byte, error) {
-	probeUtil, err := net.GetRemoteSystemProbeUtil(socketPath)
+func GetSystemProbeConntrackCached(client *http.Client) ([]byte, error) {
+	url := sysprobeclient.ModuleURL(sysconfig.NetworkTracerModule, "/debug/conntrack/cached")
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	data, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
-	return probeUtil.GetConnTrackCached()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(`GetConnTrackCached got non-success status code: url: %s, status_code: %d, response: "%s"`, req.URL, resp.StatusCode, data)
+	}
+
+	return data, nil
```

Comment:
Could we make a helper for code like this?

Reply:
I can replace these with a helper in my next PR for selinux